slang-gfx.h

```diff
@@ -1313,16 +1318,27 @@ struct RayTracingPipelineStateDesc
 class IShaderTable : public ISlangUnknown
 {
 public:
+    // Specifies the bytes to overwrite into a record in the shader table.
+    struct ShaderRecordOverwrite
+    {
+        uint32_t offset; // Offset within the shader record.
+        uint32_t size; // Number of bytes to overwrite.
+        uint8_t data[8]; // Content to overwrite.
+    };
+
     struct Desc
     {
         uint32_t rayGenShaderCount;
         const char** rayGenShaderEntryPointNames;
+        const ShaderRecordOverwrite* rayGenShaderRecordOverwrites;
```

Comment:
Why wouldn't these use the same `extendedDesc` mechanism used below? I suppose it doesn't matter for now.

When we get around to doing cleanup and renaming passes, it is going to be really important to think about how we want to handle this stuff. We have the luxury of seeing how the extensibility mechanisms in VK have fared in the wild, so we can avoid some of the same mistakes (e.g., see how almost all the bit-mask stuff in VK is at or dangerously close to the maximum number of bits even *after* they expanded most of the masks to 64 bits!).